tests/integration/targets/test_a_module/runme.yml

```diff
@@ -0,0 +1,25 @@
+- hosts: localhost
+  tasks:
+    - name: Test a_module
+      assert:
+        that:
+          # Modules/actions that do not exist
+          - "'foo_bar' is not community.general.a_module"
+          - "'foo.bar.baz' is not community.general.a_module"
+          # Short name and FQCN for builtin and other collections
+          - "'file' is community.general.a_module"
+          - "'set_fact' is community.general.a_module"
+          - "'ansible.builtin.file' is community.general.a_module"
+          - "'ansible.builtin.set_fact' is community.general.a_module"
+          - "'ansible.builtin.foo_bar' is not community.general.a_module"
+          - "'community.crypto.acme_certificate' is community.general.a_module"
+          - "'community.crypto.openssl_privatekey_pipe' is community.general.a_module"
+          - "'community.crypto.foo_bar' is not community.general.a_module"
+          # Modules from this collection (that exist or not)
+          - "'community.general.ufw' is community.general.a_module"
+          - "'community.general.foooo_really_does_not_exist' is not community.general.a_module"
+          # Local module
+          - "'local_module' is community.general.a_module"
+          # Local collection module (that exist or not)
+          - "'testns.testcoll.collection_module' is community.general.a_module"
+          - "'testns.testcoll.foobar' is not community.general.a_module"
```

Comment:
@felixfontein I can create an issue about this but I don't know if it's a bug in ansible-core or in ``a_module``.

I've found that when running a list of asserts like the one above, it gets short-circuited when evaluating a tombstoned module. It works fine when it's a normal deprecation.

For example:
```yaml
- hosts: localhost
  gather_facts: no
  tasks:
    - name: Assert that modules exist
      assert:
        that:
          - item is community.general.a_module
      loop:
        - community.crypto.acme_certificate
        - community.general.ali_instance_facts
        - community.crypto.openssl_privatekey_pipe
        - community.crypto.foo_bar
      ignore_errors: true
```

Results in the following output (notice how the last two asserts did not run):
```
PLAY [localhost] **********************************************************************************************************************

TASK [Assert that modules exist] ******************************************************************************************************
ok: [localhost] => (item=community.crypto.acme_certificate) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "community.crypto.acme_certificate",
    "msg": "All assertions passed"
}
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'item is community.general.a_module' failed. The error was: [DEPRECATED]: community.general.ali_instance_facts has been removed. Use community.general.ali_instance_info instead. This feature was removed from community.general in version 3.0.0. Please update your playbooks."}
...ignoring

PLAY RECAP ****************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=1
```